### PR TITLE
Rdcc 2261 branch

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
Rdcc 2261

### Change description ###
(feat) RDCC-2261 : Adding S2S functionality #139

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by pksaxena01 — DISMISSED** [review]
APPROVED

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -25,6 +25,8 @@ spring:
           order_inserts: true
   application:
     name: RD Caseworker Ref API
+  main:
+    allow-bean-definition-overriding: true
   security:
```
Comment (SujeethKumarAri) [inline src/main/resources/application.yaml:30]: Can below 
`- serviceauthorization` be removed from 
`feign:
  allow:
    headers:
      - authorization
      - serviceauthorization`
- https://github.com/hmcts/rd-caseworker-ref-api/blob/652cc9ca2f26767154bb53b82d09e38c8e1582ff/src/main/resources/application.yaml#L106.

as we don't pass the service header from CRD to other service now ?

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/config/FeignInterceptorConfiguration.java [hunk 2]
```diff
@@ -35,6 +41,8 @@ public RequestInterceptor requestInterceptor(FeignHeaderConfig config) {
                     log.warn("{}:: FeignHeadConfiguration {}", loggingComponentName, "Failed to get request header!");
                 }
             }
+            requestTemplate.header(CaseWorkerConstants.SERVICE_AUTHORIZATION, "Bearer "
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/config/FeignInterceptorConfiguration.java]: Acknowledged, and fixed.

File: src/main/resources/application.yaml (hunk 1 above)
Comment (nitishsingla1234) [reply src/main/resources/application.yaml:30]: Good catch Sujeeth.
Removed ServiceAuth from application.yaml.

**Review by SujeethKumarAri — APPROVED** [review]

**Review by AdilOoze — DISMISSED** [review]

**Review by shreedhar-hmcts — COMMENTED** [review]
code looks good now, just test one flow once with exui/role mapping service bearer tokens.

**Review by SujeethKumarAri — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

**Pull request merged**